src/test/java/teammates/test/cases/webapi/GetFeedbackResponsesActionTest.java

```diff
@@ -20,16 +44,240 @@ protected String getRequestMethod() {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, "randomNonExistId",
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+        };
+        assertThrows(EntityNotFoundException.class,
+                () -> getAction(nonExistParams).checkAccessControl());
     }
 
+    @Test
+    protected void testAccessControl_getResponseInPreview_shouldFail() {
+        String[] inPreviewRequest = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn1InSession1InCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.PREVIEWAS, instructor1OfCourse1.getEmail(),
+        };
+        verifyCannotAccess(inPreviewRequest);
+    }
+
+    @Test
+    protected void testAccessControl_accessAcrossCourses_shouldFail() {
+
+        ______TS("student access other student's response from different course");
+        loginAsStudent(student1InCourse2.getGoogleId());
+        String[] studentAccessOtherStudentsParams = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn1InSession1InCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+        };
+        assertThrows(EntityNotFoundException.class,
+                () -> getAction(studentAccessOtherStudentsParams).checkAccessControl());
+
+        ______TS("instructor access other instructor's response from different course");
+        loginAsInstructor(instructor1OfCourse2.getGoogleId());
+        String[] instructorAccessOtherInstructorsParams = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn2InGracePeriodInCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+        };
+        verifyCannotAccess(instructorAccessOtherInstructorsParams);
+    }
+
+    private FeedbackResponsesData getFeedbackResponse(String[] params) {
+        GetFeedbackResponsesAction a = getAction(params);
+        JsonResult actualResult = getJsonResult(a);
+        assertEquals(HttpStatus.SC_OK, actualResult.getStatusCode());
+        return (FeedbackResponsesData) actualResult.getOutput();
+    }
+
+    private void verifyFeedbackResponseEquals(FeedbackResponseAttributes expected, FeedbackResponseData actual) {
+        assertEquals(expected.getId(), actual.getFeedbackResponseId());
+        assertEquals(expected.getGiver(), actual.getGiverIdentifier());
+        assertEquals(expected.getRecipient(), actual.getRecipientIdentifier());
+        assertEquals(expected.getResponseDetails().getAnswerString(), actual.getResponseDetails().getAnswerString());
+        assertEquals(expected.getResponseDetails().questionType, actual.getResponseDetails().questionType);
+        assertEquals(JsonUtils.toJson(expected.getResponseDetails()),
+                JsonUtils.toJson(actual.getResponseDetails()));
+    }
+
+    private void verifyAnswerableForInstructor(FeedbackQuestionAttributes feedbackQuestionAttributes) {
```

Comment:
What's the point of including `verifyAnswerableForInstructor` and `verifyAnswerableForStudent` here?

They are supposed to be in the production code?